test/run-test.py

```diff
@@ -35,9 +36,21 @@ def compare_structures(data1, data2, use_regex=False):
             return False
         if len(data1) != len(data2):
             return False
-        for idx in range(0, len(data1)):
-            if not compare_structures(data1[idx], data2[idx], use_regex=use_regex):
-                return False
+        data2 = copy.deepcopy(data2)
+        for idx, var1 in enumerate(data1):
+            # order in list of dicts doesn't matter
+            if isinstance(var1, dict):
+                matchFound = False
+                for var2 in data2:
+                    if compare_structures(var1, var2, use_regex=use_regex):
+                        data2.remove(var2)
```

Comment:
It's generally a bad idea to remove elements from the struct that you're iterating over while you're iterating over it. Is removing it actually necessary?

Reply:
this is removing an element and than breaking out if the loop over data2, so this shouldn't cause any issues - the iteration continues over data1 and after that interation over data2 would happen without the found element
Removing is necessary here for situations where the data1 list contains the exact same element twice and data2 has only one of that elements.